insights/v1alpha1/types_insights.go

```diff
@@ -0,0 +1,198 @@
+	// name is the name of the gatherer.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:MinLength=5
+	Name string `json:"name"`
+	// lastGatherDuration represents the time spent gathering.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([1-9][0-9]*(\\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$"
+	LastGatherDuration metav1.Duration `json:"lastGatherDuration"`
+}
+
+// insightsReport provides Insights health check report based on the most
+// recently sent Insights data.
+type InsightsReport struct {
+	// downloadedAt is the time when the last Insights report was downloaded.
+	// An empty value means that there has not been any Insights report downloaded yet and
+	// it usually appears in disconnected clusters (or clusters when the Insights data gathering is disabled).
+	// +optional
+	DownloadedAt metav1.Time `json:"downloadedAt,omitempty"`
+	// healthChecks provides basic information about active Insights health checks
+	// in a cluster.
+	// +listType=atomic
+	// +optional
+	HealthChecks []HealthCheck `json:"healthChecks,omitempty"`
+}
+
+// healthCheck represents an Insights health check attributes.
+type HealthCheck struct {
+	// description provides basic description of the healtcheck.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:MinLength=10
+	Description string `json:"description"`
+	// totalRisk of the healthcheck. Indicator of the total risk posed
+	// by the detected issue; combination of impact and likelihood. The values can be from 1 to 4,
+	// and the higher the number, the more important the issue.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=4
+	TotalRisk int32 `json:"totalRisk"`
+	// advisorURI provides the URL link to the Insights Advisor.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^https:\/\/\S+`
+	AdvisorURI string `json:"advisorURI"`
+	// state determines what the current state of the health check is.
+	// Health check is enabled by default and can be disabled
+	// by the user in the Insights advisor user interface.
+	// +kubebuilder:validation:Required
+	State HealthCheckState `json:"state"`
+}
+
+// healthCheckState provides information about the status of the
+// health check (for example, the health check may be marked as disabled by the user).
+// +kubebuilder:validation:Enum:=Enabled;Disabled
+type HealthCheckState string
+
+const (
+	// enabled marks the health check as enabled
+	HealthCheckEnabled HealthCheckState = "Enabled"
+	// disabled marks the health check as disabled
+	HealthCheckDisabled HealthCheckState = "Disabled"
+)
+
+// ObjectReference contains enough information to let you inspect or modify the referred object.
+type ObjectReference struct {
+	// group of the referent.
+	// +kubebuilder:validation:Required
+	// +required
```

Comment:
Please drop `+required` as we prefer the `+kubebuilder:validation:Required` variant

Reply:
fixed